### PR TITLE
User story 24

### PR DESCRIPTION
**Pull request by NickEdwin**
For some reason it wouldn't let me push to Sienna's 23... but since 24 is almost completed just merge this. 

We can work out any conflicts later. 

Functionality completed for increasing and decreasing item count in cart.

### CODE REVIEW

**Comment by NickEdwin** [comment]
Not *almost completed, 24 IS completed.

File: app/controllers/cart_controller.rb [hunk 1]
```diff
@@ -23,6 +23,27 @@ def remove_item
     redirect_to '/cart'
   end
 
+
+  def increase
```
Comment (sienna-kopf) [inline app/controllers/cart_controller.rb:27]: Clever method names.

File: app/controllers/cart_controller.rb [hunk 2]
```diff
@@ -23,6 +23,27 @@ def remove_item
...
+    item = Item.find(params[:item_id])
+    unless item.inventory == cart.contents[params[:item_id]]
```
Comment (sienna-kopf) [inline app/controllers/cart_controller.rb:29]: Clever check here to make sure inventory isn't increased past the inventory amount

File: app/controllers/cart_controller.rb [hunk 3]
```diff
@@ -23,6 +23,27 @@ def remove_item
...
+      flash[:error] = "You have exceeded the maximum inventory for #{item.name}!"
+    end
+    redirect_back(fallback_location: root_path)
+  end
+
+
+  def decrease
+    item = Item.find(params[:item_id])
+    cart.contents[params[:item_id]] -= 1
+    if cart.contents[params[:item_id]] == 0
```
Comment (sienna-kopf) [inline app/controllers/cart_controller.rb:41]: Nice check here to make sure it's not depleted past zero.

File: app/views/cart/show.html.erb [hunk 4]
```diff
@@ -31,12 +31,14 @@
       <td>
         <center>
           <p><%=quantity%></p>
+          <p><%= link_to "Increase Amount", "/cart/#{item.id}/increase", method: :put %></p>
```
Comment (sienna-kopf) [inline app/views/cart/show.html.erb:34]: the fact that these were put methods was key

**Pull request merged**